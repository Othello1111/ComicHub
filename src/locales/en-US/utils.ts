export default {
  'utils.request.status.error': 'Request Fail',
  'utils.request.100': 'Continue',
  'utils.request.200': 'Request Success',
  'utils.request.201': 'Created',
  'utils.request.202': 'Accepted',
  'utils.request.204': 'No Content',
  'utils.request.400': 'Bad Request',
  'utils.request.401': 'Unauthorized',
  'utils.request.403': 'Forbidden',
  'utils.request.404': 'Not Found',
  'utils.request.406': 'Not Acceptable',
  'utils.request.408': 'Request Timeout',
  'utils.request.410': 'Gone',
  'utils.request.422': 'Unprocessable Entity',
  'utils.request.500': 'Internal Server Error',
  'utils.request.501': 'Not Implemented',
  'utils.request.502': 'Bad Gateway',
  'utils.request.503': 'Service Unavaliable',
  'utils.request.504': 'Gateway Timeout',
};
