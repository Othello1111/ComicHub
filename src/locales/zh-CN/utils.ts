export default {
  'utils.request.status.error': '请求错误',
  'utils.request.100': '请求头已接受成功，请求数据待发送。',
  'utils.request.200': '服务器成功返回请求的数据。',
  'utils.request.201': '新建或修改数据成功。',
  'utils.request.202': '一个请求已经进入后台排队（异步任务）。',
  'utils.request.204': '删除数据成功。',
  'utils.request.400': '发出的请求有错误，服务器没有进行新建或修改数据的操作。',
  'utils.request.401': '用户没有权限（令牌、用户名、密码错误）。',
  'utils.request.403': '用户得到授权，但是访问是被禁止的。',
  'utils.request.404': '发出的请求针对的是不存在的记录，服务器没有进行操作。',
  'utils.request.406': '请求的格式不可得。',
  'utils.request.408': '服务器关闭连接。',
  'utils.request.410': '请求的资源被永久删除，且不会再得到的。',
  'utils.request.422': '当创建一个对象时，发生一个验证错误。',
  'utils.request.500': '服务器发生错误，请检查服务器。',
  'utils.request.501': '服务器不支持请求方法。',
  'utils.request.502': '网关错误。',
  'utils.request.503': '服务不可用，服务器暂时过载或维护。',
  'utils.request.504': '网关超时。',
};
