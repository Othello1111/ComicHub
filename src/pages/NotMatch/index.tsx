import React from 'react';
import { Link } from 'react-router-dom';
import { Location } from 'history';

type Props = {
  location: Location;
};

const NoMatch: React.FunctionComponent<Props> = ({
  location,
}) => {
  return (
    <div>
      <h3>
        No match for <code>{location.pathname}</code>
      </h3>
      <div>
        <Link to="/">Back To Home</Link>
      </div>
    </div>
  );
};

export default NoMatch;
