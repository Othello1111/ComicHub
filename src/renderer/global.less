@import '~antd/lib/style/themes/default.less';

body {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #000;
  line-height: 1.5;
  background-color: #f2f4f5;
}
