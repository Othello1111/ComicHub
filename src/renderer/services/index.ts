import request from '../utils/request';

interface PostData {
    url?: string;

    [index: string]: string;
}

export async function getMenuList() {
    return request('/v1/menu');
}

export async function postItem(params: PostData = {}) {
    const { url, ...rest } = params;
    if (!url) {
        return;
    }
    const realUrl = `/v1/${url}`;
    return request(realUrl, {
        method: 'POST',
        body: rest,
    });
}
