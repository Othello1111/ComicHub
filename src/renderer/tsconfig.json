{
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        "importHelpers": true,
        "jsx": "preserve",
        "esModuleInterop": true,
        "sourceMap": true,
        "strict": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "baseUrl": ".",
        "paths": {
            "@/*": ["./*"]
        },
        "allowSyntheticDefaultImports": true
    }
}
