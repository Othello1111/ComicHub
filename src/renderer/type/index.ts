export interface SharedState {
    currentUrl: string;
    currentType: string;
}
export interface MenuItem {
    value: string;
    enabled: boolean;
    base: string;
    name: string;
}
export interface IFormData {
    url: string;
    name: string;
}

export interface IOptionData {
    value: string | number;
    name: string;
    enabled: boolean;
}
