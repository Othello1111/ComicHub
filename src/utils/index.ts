export function addZero(temp: number): string {
    if (temp < 10) {
        return `0${temp}`;
    }
    return `${temp}`;
}

/**
 * 统一格式化时间
 * @param temp
 */
export function renderDate(temp: number | string): string {
    let date: Date;
    try {
        date = new Date(temp);
    } catch (e) {
        return '';
    }
    const year: number = date.getFullYear();
    if (Number.isNaN(year)) {
        return '';
    }
    const month: number = date.getMonth() + 1;
    const day: number = date.getDate();
    const hour: number = date.getHours();
    const minute: number = date.getMinutes();
    const second: number = date.getSeconds();
    return `${year}-${addZero(month)}-${addZero(day)} ${addZero(hour)}:${addZero(minute)}:${addZero(second)}`;
}


export enum typeConfig {
    chapter = 'chapter',
    downloadAll = 'downloadAll',
    download = 'images',
    search = 'search',
    result = 'result',
}
