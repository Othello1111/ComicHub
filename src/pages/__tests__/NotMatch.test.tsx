import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { history } from '../../utils';
import NotMatch from '../NotMatch';

it('Page: Chpter  Snapshots', () => {
  const location = { pathname: 'test', search: '', hash: '', state: '' };

  let tree: ReactTestRenderer;
  renderer.act(() => {
    tree = renderer
      .create(<Router history={history}><NotMatch location={location}/></Router>);
  });
  // @ts-ignore
  expect(tree.toJSON()).toMatchSnapshot();
});
