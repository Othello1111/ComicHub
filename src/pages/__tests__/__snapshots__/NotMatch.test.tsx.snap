// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Chpter  Snapshots 1`] = `
<div>
  <h3>
    No match for 
    <code>
      test
    </code>
  </h3>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      Back To Home
    </a>
  </div>
</div>
`;
